rust/ql/test/query-tests/unusedentities/unreachable.rs

```diff
@@ -232,22 +352,32 @@ pub fn unreachable_let_2() {
     do_something();
 }
 
-pub fn unreachable_if_2() {
-    if cond() {
+#[cfg(not(foo))]
+pub fn unreachable_attributes() {
+    // `#[cfg` and `cfg!` checks can go either way, we should not assume this
+    // function or either branch below is unreachable.
+    if cfg!(bar) {
         do_something();
-        return;
     } else {
         do_something();
     }
 
-    do_something();
-}
+    #[doc="This is a doc comment declared through an attribute."]
 
-pub fn unreachable_if_3() {
-    if !cond() {
+    if (true) {
         do_something();
-        return;
+    } else {
+        do_something(); // $ Alert[rust/dead-code]
     }
-
-    do_something();
 }
+
+const _: () = {
+    _ = 1; // $ SPURIOUS: Alert[rust/dead-code]
```

Comment:
One argument for including them in the CFG is that `const` code could contain unreachable code. If we exclude constant things, then we will not be able to detect this and other things (unused variables, etc.). An argument in the other direction is that code evaluated at compile time is unlikely to contain security issues. Or maybe it can?

Reply:
It feels like `const` code is _less_ likely to contain security issues, but still could.  Certainly for data flow - for example the value of a security relevant constant (e.g. enabling or disabling a feature) could be computed in `const` code.

Reply:
Do you know what steps we need to take to include them in the CFG?  Is it just a case of adding the entry points somewhere???

Reply:
I've created an issue to track this, but I could do with some pointers (or someone else to take on the issue, if that is easier).

Reply:
Thanks! I'll be looking at that together as well as the `async` problem 👍